ui/app/AppMain.qml

```diff
@@ -350,76 +350,49 @@ Item {
         appView: StackLayout {
             id: appView
             anchors.fill: parent
-            currentIndex: {
-                if(mainModule.activeSection.sectionType === Constants.appSection.chat) {
-                    return Constants.appViewStackIndex.chat
-                }
-                else if(mainModule.activeSection.sectionType === Constants.appSection.community) {
-
-                    for(let i = this.children.length - 1; i >=0; i--)
-                    {
-                        var obj = this.children[i];
-                        if(obj && obj.sectionId == mainModule.activeSection.id)
-                        {
-                            return i
-                        }
-                    }
-
-                    // Should never be here, correct index must be returned from the for loop above
-                    console.error("Wrong section type: ", mainModule.activeSection.sectionType,
-                                  " or section id: ", mainModule.activeSection.id)
-                    return Constants.appViewStackIndex.community
-                }
-                else if(mainModule.activeSection.sectionType === Constants.appSection.wallet) {
-                    return Constants.appViewStackIndex.wallet
-                }
-                else if(mainModule.activeSection.sectionType === Constants.appSection.walletv2) {
-                    return Constants.appViewStackIndex.walletv2
-                }
-                else if(mainModule.activeSection.sectionType === Constants.appSection.browser) {
-                    return Constants.appViewStackIndex.browser
-                }
-                else if(mainModule.activeSection.sectionType === Constants.appSection.timeline) {
-                    return Constants.appViewStackIndex.timeline
-                }
-                else if(mainModule.activeSection.sectionType === Constants.appSection.profile) {
-                    return Constants.appViewStackIndex.profile
-                }
-                else if(mainModule.activeSection.sectionType === Constants.appSection.node) {
-                    return Constants.appViewStackIndex.node
-                }
-
-                // We should never end up here
-                console.error("Unknown section type")
-            }
-
+            currentIndex: appMain.rootStore.appViewIndex
```

Comment:
@alexandraB99 `appMain.rootStore.appViewIndex` you used here refers to the section type, not to the index of a component in this `StackLayout`.